### ISSUE TITLE
Multinomial Coefficient 

### ISSUE THREAD

**Issue by Tengs-Penkwe** [typst/typst#2008]
### Description

Typst doesn't support Multinomial Coefficient
`$binom(n, (n_1, n_2, ...))$` gives 
<img width="73" alt="Screenshot 2023-08-25 at 16 55 16" src="https://github.com/typst/typst/assets/61036442/35e9e57f-2460-49a0-b09d-7e1c60bd3f6f">

### Use Case

Like latex
`$\binom{n}{n_1, n_2, \dots}$`
$\binom{n}{n_1, n_2, \dots}$

**Comment by xkevio** [comment]
It does but you have to escape the commas with a backslash so it gets interpreted as a symbol and not as a way to specify an additional parameter. Then you can also drop the parentheses.

**Comment by Tengs-Penkwe** [comment]
> It does but you have to escape the commas with a backslash so it gets interpreted as a symbol and not as a way to specify an additional parameter. Then you can also drop the parentheses.

Thanks!

**Comment by Enivex** [comment]
You can also do something like

```#let binomx(..x) = math.binom(x.pos().first(),x.pos().slice(1).join(","))```

Then you can include as many argument as you'd like, and they'll be separated by commas

**Comment by laurmaedje** [comment]
@Enivex Would that be reasonable default behaviour? I had never heard of multinomial coefficients before.

**Comment by Enivex** [comment]
> @Enivex Would that be reasonable default behaviour? I had never heard of multinomial coefficients before.

I would say that it's fairly reasonable, but @damaxwell can chime in as well.

**Comment by damaxwell** [comment]
Sounds reasonable to me to extend `binom` this way by default; I can't think of an alternative use for three or more arguments.
  
But the pain point still exists generally: things with commas show up with some low but nonzero frequency as single arguments to functions in math and there isn't a super natural way to just group the comma-ed things into one blob like you would with `{` `}` in TeX.  I was surprised (at first) that `$ binom(a, $b, c, d$) $` doesn't work.

**Comment by Enivex** [comment]
> I was surprised (at first) that `$ binom(a, $b, c, d$) $` doesn't work.

That does sort of work, you just need a # in front of the $

**Comment by damaxwell** [comment]
Ah, yes.  That's a decent workaround, actually.

**Comment by Doublonmousse** [comment]
I wonder if the `{` `}` syntax from tex could be added as a optional method for grouping elements that is stronger than the `(` `)` syntax.

This way, we both have a more convenient way to group elements with `(`, `)` most of the time, and `{`, `}` for these other cases that needs escaping or other workarounds.

**Comment by TheJosefOlsson** [comment]
There are several suggestions I've seen, but which is preferred? No brackets and using only commas? Brackets on the denominator? Curly brackets? Or should I just choose whichever I think is best when solving the issue?

Also, during testing I thought about this case, is this intended?

```
#let f(x) = (
    x * x
)

$binom(n, (#f(4)))$ // works but yields (16)
// $binom(n, #f(4))$ // does not work
```

**Comment by laurmaedje** [comment]
The grouping issue discussed above is separate, so I'd say just commas (i.e. just more arguments to `binom`).

The case you found indeed feels a bit wrong, but it is also a more general thing (unrelated to binom, sqrt gives you the same error). We could perhaps have a separate issue for it.

**Comment by TheJosefOlsson** [comment]
So the syntax should be `$binom(n, n_1, n_2, ...)$`?

Is it reasonable to make my own issue for the other strange behaviour?

**Comment by laurmaedje** [comment]
> So the syntax should be `$binom(n, n_1, n_2, ...)$?`

Yep

> Is it reasonable to make my own issue for the other strange behaviour?

Yeah, feel free to do that.

**Comment by TheJosefOlsson** [comment]
I have not found a way to add a span that annotates the missing argument the same way as when it was not variadic and manually parsed. One of the problems is that the only span that I can find that is available is the spanned values, but they will not exist since we're checking to see if there are no values. I would need to access the span of the binomial itself, which I can't understand how to do. That is why I updated the test the way I did
